yroots/subdivision.py

```diff
@@ -99,61 +99,60 @@ def solve(funcs, a, b, rel_approx_tol=1.e-15, abs_approx_tol=1.e-12,
     -------
     zeros : numpy array
         The common zeros of the polynomials. Each row is a root.
-    '''
-    #Detect the dimension
+    """
+    # Detect the dimension
     if not isinstance(funcs,list):
         dim = 1
     else:
         dim = len(funcs)
 
-    #make a and b the right type
+    # make a and b the right type
     a = np.float64(a)
     b = np.float64(b)
 
     # Choose an appropriate max degree for the given dimension if none is specified.
     if deg is None:
-        deg_dim = {1: 50, 2:9, 3:5, 4:3}
+        deg_dim = {1: 100, 2:20, 3:9, 4:9}
```

Comment:
I'm assuming these are the new values based on the tests you did, but I think I was gone when this was decided.

Reply:
The values for 1D and 2D are from experimental data, but for dimensions 3 and 4 I just defaulted back to degree 9. Before I changed things on this Pull Request, we were using degree 20 approximations for 3 and 4 dimensions, which slowed down our solver significantly (on the random 3D and 4D polynomial tests as part of our PyTests).

I think we still should look into what degrees to default to for 3 and 4.